nasa/build-env.sh

```diff
@@ -1,9 +1,19 @@
-#!/bin/bash
+#!/usr/bin/env bash
 
-basedir=$( cd "$(dirname "$0")" ; pwd -P )
-REPO_ROOT_PATH=$(dirname ${basedir})
+set -xeou pipefail
 
-conda config --set solver libmamba
+basedir=$(dirname "$(dirname "$(readlink -f "$0")")")
+conda=${CONDA_EXE:-conda}
 
-conda env create -f ${REPO_ROOT_PATH}/environment.yaml
+PIP_REQUIRE_VENV=0 "${conda}" env update --quiet --prune --solver libmamba \
+    --name dem --file "${basedir}"/environment.yml
 
+if [ "${1:-}" == "--dev" ]; then
+    # Install development dependencies
+    PIP_REQUIRE_VENV=0 "${conda}" env update --quiet --solver libmamba \
+        --name dem --file "${basedir}"/environment-dev.yml
+    # Install the kernel for Jupyter
```

Comment:
Correct. This is not installed for DPS. This is why it is wrapped in a conditional. This is not documented in the README yet, but a developer would run `nasa/build-env.sh --dev` to get the kernel installed as well as the dev dependencies. For DPS, the build command is simply `nasa/build-env.sh` (see `nasa/algorithm_config.yml`), so the kernel and dev deps are not installed for a DPS job.

I didn't add details to the README yet, because I wanted to get a minimal set of changes committed for ESA to try to leverage. My thinking is that once we get kinks ironed out to the point that we have everything working for NASA and ESA, we could then add the appropriate documentation.

Reply:
Background -- In the time since these comments were originally added, the `environment-dev.yml`  was moved into the `nasa` directory. (See: https://github.com/MAAP-Project/get-dem/pull/6#discussion_r1559682538)

Since this branch is working for NASA MAAP, is now the right time to update the README with a new a NASA-only section describing this?